rapid7/metasploit-credential
# Create credential and login

We often create a login immediately after creating a credential. I'm rolling the two steps into a single method call. 

### Verification steps
open a rails console in ```spec/dummy```
```
workspace = FactoryGirl.create(:mdm_workspace)
user = FactoryGirl.create(:mdm_user)
service = FactoryGirl.create(:mdm_service, host: FactoryGirl.create(:mdm_host, workspace: workspace))

login_data = {
  workspace_id: workspace.id,
  origin_type: :manual,
  user_id: user.id,
  username: 'admin',
  private_data: 'password',
  private_type: :password,
  workspace_id: workspace.id,
  address: service.host.address,
  port: service.port,
  service_name: service.name,
  protocol: service.proto,
  last_attempted_at: DateTime.current,
  status: Metasploit::Model::Login::Status::SUCCESSFUL,
}
test_object = Class.new do
  include Metasploit::Credential::Creation
end.new

test_object.create_credential_and_login(login_data)
```
that should create the following objects
```
core = Metasploit::Credential::Core.last
core.public.username == 'admin'
core.private.data == 'password'
core.logins.first.status == 'Successful'
core.logins.first.service == service
```
